app/src/main/java/com/github/fribourgsdp/radio/LanguageManager.kt

```diff
@@ -0,0 +1,42 @@
+package com.github.fribourgsdp.radio
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.content.res.Configuration
+import android.content.res.Resources
+import android.util.DisplayMetrics
+import java.util.*
+
+class LanguageManager {
+
+    private lateinit var context: Context
+    private lateinit var sharedPreferences: SharedPreferences
+
+    constructor(context : Context){
+        this.context = context
+        sharedPreferences = context.getSharedPreferences("LANG", Context.MODE_PRIVATE)
+    }
+
+    fun updateResource(code: String){
+        val locale = Locale(code)
+        Locale.setDefault(locale)
+        val resources : Resources = context.getResources()
+        val metrics : DisplayMetrics = resources.displayMetrics
+        val configuration : Configuration = resources.configuration
+        configuration.setLocale(locale)
+        resources.updateConfiguration(configuration, metrics)
+        setLang(code)
+    }
+
+    fun getLang() : String? {
+        return sharedPreferences.getString("lang", "en")
+    }
+
+    fun setLang(code :String){
+        val editor : SharedPreferences.Editor = sharedPreferences.edit()
+        editor.putString("lang", code)
+        editor.commit()
```

Comment:
Here there is a warning on Cirrus that says you should use `apply` instead of `commit` so that it doesn't need to write the data to the main memory right away all the time. I let you consider if it is a good thing or not.

Reply:
Yes, I had seen it too. As a user will only change the language once in a while, there is no point in using apply and you can commit directly.